### PR TITLE
feat(icons): added calendar-cog icon

### PR DESCRIPTION
**Pull request by AlexandrePhilibert**
## What is the purpose of this pull request?

- [x] New Icon
- [ ] Bug fix
- [ ] New Feature
- [ ] Documentation update
- [ ] Other:

### Description

Adds a calendar with a cog in the bottom right.

### Icon use case 

Can be used in the following circumstances:
- Settings related to a calendar
- Settings related to events

### Concept 

- [x] I have provided valid use cases for each icon.
- [x] I have not added any a brand or logo icon.
- [x] I have not used any hate symbols.
- [x] I have not included any religious or political imagery.

### Author, credits & license

- [ ] The icons are solely my own creation.
- [ ] The icons were originally created in #<issueNumber> by @<githubUser>
- [x] I've based them on the following Lucide icons: 
  - [user-cog](https://lucide.dev/icons/user-cog)
  - [calendar-search](https://lucide.dev/icons/calendar-search)
- [ ] I've based them on the following design: 

### Naming 

- [x] I've read and followed the [naming conventions](https://lucide.dev/guide/design/icon-design-guide#naming-conventions)
- [x] I've named icons by what they are rather than their use case.
- [x] I've provided meta JSON files in `icons/[iconName].json`.

### Design 

- [x] I've read and followed the [icon design guidelines](https://lucide.dev/guide/design/icon-design-guide)
- [x] I've made sure that the icons look sharp on low DPI displays.
- [x] I've made sure that the icons look consistent with the icon set in size, optical volume and density.
- [x] I've made sure that the icons are visually centered.
- [x] I've correctly optimized all icons to three points of precision.

## Before Submitting 

- [x] I've read the [Contribution Guidelines](https://github.com/lucide-icons/lucide/blob/main/CONTRIBUTING.md).
- [x] I've checked if there was an existing PR that solves the same issue.

### CODE REVIEW

**Comment by jguddas** [comment]
Cleaned up

<a title="Open lucide studio" target="_blank" href="https://lucide-studio.vercel.app/edit?value=%3Cpath+d=%22m15.2+16.9-.9-.4%22+/%3E%3Cpath+d=%22m15.2+19.1-.9.4%22+/%3E%3Cpath+d=%22M16+2v4%22+/%3E%3Cpath+d=%22m16.9+15.2-.4-.9%22+/%3E%3Cpath+d=%22m16.9+20.8-.4.9%22+/%3E%3Cpath+d=%22m19.5+14.3-.4.9%22+/%3E%3Cpath+d=%22m19.5+21.7-.4-.9%22+/%3E%3Cpath+d=%22M21+10.5V6a2+2+0+0+0-2-2H5a2+2+0+0+0-2+2v14a2+2+0+0+0+2+2h6%22+/%3E%3Cpath+d=%22m21.7+16.5-.9.4%22+/%3E%3Cpath+d=%22m21.7+19.5-.9-.4%22+/%3E%3Cpath+d=%22M3+10h18%22+/%3E%3Cpath+d=%22M8+2v4%22+/%3E%3Ccircle+cx=%2218%22+cy=%2218%22+r=%223%22+/%3E">icons<br/>Open lucide studio</a>

**Review by jguddas — CHANGES_REQUESTED** [review]
See comment.

**Comment by AlexandrePhilibert** [comment]
@jguddas I've applied you changes, let me know if there's anything else I need to do.

File: icons/calendar-cog.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+  "$schema": "../icon.schema.json",
+  "contributors": [
+    "karsa-mistmere",
```
Comment (jguddas) [inline icons/calendar-cog.json:4]: Can you add the people from the original calendar and cog icon you used here?

Comment (AlexandrePhilibert) [reply icons/calendar-cog.json:4]: I have already added karsa-mistmere and ericfennis that are the contributors to the two icons I based mine on.

**Review by ericfennis — APPROVED** [review]
Nice work @AlexandrePhilibert , love it!

**Review by karsa-mistmere — APPROVED** [review]
LGTM!

File: icons/calendar-cog.json (hunk 1 above)
Comment (jguddas) [reply icons/calendar-cog.json:4]: Ohh, looks like I messed up and didn't add myself when I redid the cogs in #1225

**Review by jguddas — APPROVED** [review]

**Pull request merged**